dao/tenant_dao.go

```diff
@@ -20,28 +22,62 @@ func init() {
 
 type tenantDaoImpl struct{}
 
-func (t *tenantDaoImpl) GetOrCreateTenantID(accountNumber string) (*int64, error) {
-	tenant := m.Tenant{ExternalTenant: accountNumber}
+func (t *tenantDaoImpl) GetOrCreateTenantID(identity *identity.Identity) (int64, error) {
+	// Start setting up the query.
+	query := DB.
+		Debug().
+		Model(&m.Tenant{})
+
+	// Query by OrgId or EBS account number.
+	var tenant m.Tenant
+	if identity.OrgID != "" {
+		tenant.OrgID = identity.OrgID
+
+		query.Where("org_id = ?", tenant.OrgID)
+	} else {
+		tenant.ExternalTenant = identity.AccountNumber
+
+		query.Where("external_tenant = ?", tenant.ExternalTenant)
+	}
```

Comment:
I had to add a condition so that it checks we don't fetch the ones that have an empty `external_tenant` or `org_id`, since otherwise we could fetch the wrong tenant.